cmd/earthly/app/run.go

```diff
@@ -384,3 +383,24 @@ func (app *EarthlyApp) printCrashLogs(ctx context.Context) {
 func errorWithPrefix(err string) string {
 	return fmt.Sprintf("Error: %s", err)
 }
+
+func getHintErr(err error, grpcError *status.Status, target **hint.Error) bool {
```

Comment:
the double pointer here seems complicated; is there a reason we can't always return a new `hint.Error` instead?

e.g.

```
func getHintErr(err error, grpcError *status.Status) (hint.Error, bool) {
```

or

```
func getHintErr(err error, grpcError *status.Status) (*hint.Error, bool) {
```

?

Reply:
or maybe even just

```
func getHintErr(err error) (hint.Error, bool) {
```

and have it call `grpcErr, grpcErrOK := grpcerrors.AsGRPCStatus(err)` inside the function?

Reply:
Yeah, it's so I can use it in the switch case condition (where it's called.
I could refactor and call it out of the condition.

side note: I was thinking of refactoring this whole switch case to make it much easier to maintain/read (hopefully), 
that should make these kind of considerations go away.

Reply:
> and have it call grpcErr, grpcErrOK := grpcerrors.AsGRPCStatus(err) inside the function?

This is already called outside the function so I didn't want to call it twice.
This is also something that should be improved with the refactor I mentioned.

Reply:
done